lab_5_scrapper/scrapper.py

```diff
@@ -2,73 +2,172 @@
+        self._encoding = self._config_data.encoding
+        self._timeout = self._config_data.timeout
+        self._verify_certificate = self._config_data.should_verify_certificate
+        self._headless_mode = self._config_data.headless_mode
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            content = json.load(f)
+        return ConfigDTO(**content)
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            content = json.load(f)
+
+        seed_urls = content['seed_urls']
+        headers = content['headers']
+        total_articles_to_find_and_parse = content['total_articles_to_find_and_parse']
+        encoding = content['encoding']
+        timeout = content['timeout']
+        should_verify_certificate = content['should_verify_certificate']
+
+        if not isinstance(seed_urls, list)\
+                or not not all(isinstance(url, str) for url in seed_urls)\
+                or not all(re.match('https?://w?w?w?.', url) for url in seed_urls):
+            raise IncorrectSeedURLError
+
+        if not isinstance(headers, dict):
+            raise IncorrectHeadersError
+
+        if not isinstance(encoding, str):
+            raise IncorrectEncodingError
+
+        if not isinstance(total_articles_to_find_and_parse, int):
+            raise IncorrectNumberOfArticlesError
+
+        if total_articles_to_find_and_parse not in range(1, 151):
+            raise NumberOfArticlesOutOfRangeError
+
+        if not isinstance(timeout, int) or not timeout < 60 or not timeout > 0:
+            raise IncorrectTimeoutError
+
+        if not isinstance(should_verify_certificate, bool):
+            raise IncorrectVerifyError
 
     def get_seed_urls(self) -> list[str]:
         """
         Retrieve seed urls
         """
-        pass
+        return self._seed_urls
 
     def get_num_articles(self) -> int:
         """
         Retrieve total number of articles to scrape
         """
-        pass
+        return self._num_articles
 
     def get_headers(self) -> dict[str, str]:
         """
         Retrieve headers to use during requesting
         """
-        pass
+        return self._headers
 
     def get_encoding(self) -> str:
         """
         Retrieve encoding to use during parsing
         """
-        pass
+        return self._encoding
 
     def get_timeout(self) -> int:
         """
         Retrieve number of seconds to wait for response
         """
-        pass
+        return self._timeout
 
     def get_verify_certificate(self) -> bool:
         """
         Retrieve whether to verify certificate
         """
-        pass
+        return self.get_verify_certificate()
```

Comment:
this is recursive, what will be returned?

Reply:
remade it